src/cms/views/media/media_actions.py

```diff
@@ -216,6 +219,71 @@ def edit_file_ajax(request, region_slug=None):
         }
     )
 
+@require_POST
+@login_required
+@region_permission_required
+@permission_required("cms.replace_mediafile")
+@json_response
+# pylint: disable=unused-argument
+def replace_file_ajax(request, region_slug=None):
+    """
+    View provides the replacement of a file via AJAX.
+
+    :param request: The current request
+    :type request: ~django.http.HttpRequest
+
+    :param region_slug: The slug of the current region
+    :type region_slug: str
+
+    :return: JSON response which indicates error or success
+    :rtype: ~django.http.JsonResponse
+    """
+    region = Region.get_current_region(request)
+
+    media_file = get_object_or_404(
+        MediaFile.objects.filter(region=region), id=request.POST.get("id")
+    )
+    logger.debug(request.FILES)
+    media_file_form = ReplaceMediaFileForm(data=request.POST, instance=media_file, files=request.FILES)
+
+    if not media_file_form.is_valid():
+        return JsonResponse(
+            {
+                "messages": media_file_form.get_error_messages(),
+            },
+            status=400,
+        )
+
+    if not media_file_form.has_changed():
+        return JsonResponse(
+            {
+                "messages": [
+                    {
+                        "type": "info",
+                        "text": _("No changes detected"),
+                    }
+                ],
+                "file": media_file.serialize(),
+            }
+        )
+
+    # Save form
+    media_file_form.save()
+
```

Comment:
```suggestion
    logger.info("%r was replaced by %r", media_file, request.user)

```
Maybe add a quick notice that a file was replaced, this might help debug problems on the production environment later on.